src/Scheduler/GenericScheduler.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use DateTime;
+use Tempest\Console\ConsoleCommand;
+use Tempest\Console\ConsoleOutput;
+
+final class GenericScheduler implements Scheduler
+{
+    public function __construct(
+        private SchedulerConfig $config,
+        private ConsoleOutput $output,
+        private ScheduledCommandsResolver $resolver,
+    ) {
+    }
+
+    public function run(?DateTime $date = null): void
+    {
+        $date ??= new DateTime();
+
+        $commands = $this->resolver->resolve($date);
+
+        foreach ($commands as $command) {
+            $this->execute($command);
+        }
+    }
+
+    private function execute(ConsoleCommand $scheduledCommand): void
+    {
+        $name = $scheduledCommand->getName();
+        $this->output->writeln("Running command: $name");
```

Comment:
I think I would prefer to have this scheduler not worry about writing output. Instead it should dispatch events that `schedule:run` hooks into, to write to the output. 

There's one big caveat though: events aren't included in `tempest/core` yet 😅 So feel free to ignore this comment, I'll fix it after merging this PR :)

Reply:
Yeah for sure - this entire PR was more so to get directions, not merge it yet :D